### PR TITLE
OpenTracing support

### PR DESCRIPTION
**Pull request by rvowles**
# Description

Start of the Interceptor support for the SDKs, updated documentation and Open Tracing support. Lock status is now being passed to SDK.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

### CODE REVIEW

File: docs/index.adoc [hunk 1]
```diff
@@ -92,36 +92,56 @@ Choose from your development language / framework and follow the links for the i
...
 |Block until Start|N|N|Y|N|N
 |Readyness Listeners|Y|Y|Y|Y|Y
 |Feature Listeners|Y|Y|Y|Y|Y
 |Feature Listener Removal|N|N|Y|Y|Y
-|Catch & Release|N|Y|N|Y|N
 |Analytics Support|Y|Y|N|Y|N
 |Google Analytics|Y|Y|N|N|N
+|===================================
+
+==== UI focused capabilities for SDKs
```
Comment (IrinaSouth) [inline docs/index.adoc:117]: maybe better to say Web focused?

File: docs/index.adoc [hunk 2]
```diff
@@ -92,36 +92,56 @@ Choose from your development language / framework and follow the links for the i
...
 - *(1)* Javascript and Typescript are supported via a Typescript libraries.
 This is available https://www.npmjs.com/org/featurehub[ at the npm repository].
 - *(2)* Dart and Flutter are supported by Dart libraries available at https://pub.dev/publishers/featurehub.io/packages[pub.dev].
 - *(3)* Java is supported by libraries from Apache Maven Central. [needs ref]
 - *(4)* C# and .NET is supported by libraries from nuget. [needs ref]
 
-=== Capabilities explained
+=== Run Capabilities
+
+The following capbilities are focused around general runtime of your application, be it a client or server based application.
```
Comment (IrinaSouth) [inline docs/index.adoc:144]: typo in "capabilities"

File: docs/index.adoc [hunk 3]
```diff
@@ -171,6 +184,22 @@ We intend over time to support only one, where the data is posted to a backend s
 This is a client side implementation of the Analytics support.
 It is designed so you need to specify the `User-ID` or `CID`, your `UA-` id and when logging an event, it will fire off into GA the event - one for each value of non-JSON features.
 
+=== UI Capabilities
```
Comment (IrinaSouth) [inline docs/index.adoc:187]: same, is it better to say Web ?

File: examples/todo-backend-java/src/main/java/todo/backend/Application.java [hunk 4]
```diff
@@ -61,6 +69,8 @@ public Application() {
 
     final HttpServer server = GrizzlyHttpServerFactory.createHttpServer(BASE_URI, config, false);
 
+    // call "server.start()" here if you wish to start the application without waiting for features
+    log.info("Waiting on complete features before starting.");
```
Comment (IrinaSouth) [inline examples/todo-backend-java/src/main/java/todo/backend/Application.java:73]: complete features?

File: sdks/java/client-java-core/src/main/java/io/featurehub/client/FeatureContext.java [hunk 5]
```diff
@@ -44,7 +44,7 @@
    * instance this is running on. The state of all features are passed.
    *
    * @param action - passed directly as a name or key
-   * @param other - any other daya (or null)
+   * @param other  - any other daya (or null)
```
Comment (IrinaSouth) [inline sdks/java/client-java-core/src/main/java/io/featurehub/client/FeatureContext.java:47]: daya?

File: docs/index.adoc (hunk 1 above)
Comment (rvowles) [reply docs/index.adoc:117]: it is used in Dart as well, which reminds me!

File: examples/todo-backend-java/src/main/java/todo/backend/Application.java (hunk 4 above)
Comment (rvowles) [reply examples/todo-backend-java/src/main/java/todo/backend/Application.java:73]: i.e. all of the features

File: sdks/java/client-java-core/src/main/java/io/featurehub/client/FeatureContext.java (hunk 5 above)
Comment (rvowles) [reply sdks/java/client-java-core/src/main/java/io/featurehub/client/FeatureContext.java:47]: data

**Pull request merged**